src/Synapse/Synapse/Commands/DataPlaneCommands/AccessControl/GetAzureSynapseRoleAssignment.cs

```diff
@@ -0,0 +1,154 @@
+            Mandatory = false, HelpMessage = HelpMessages.SignInName)]
+        [Parameter(ValueFromPipelineByPropertyName = false, ParameterSetName = GetByWorkspaceObjectAndNameParameterSet,
+            Mandatory = true, HelpMessage = HelpMessages.SignInName)]
+        [Alias("Email", "UserPrincipalName")]
+        [ValidateNotNullOrEmpty]
+        public string SignInName { get; set; }
+
+        [Parameter(ValueFromPipelineByPropertyName = false, ParameterSetName = GetByWorkspaceNameAndServicePrincipalNameParameterSet,
+            Mandatory = false, HelpMessage = HelpMessages.ServicePrincipalName)]
+        [Parameter(ValueFromPipelineByPropertyName = false, ParameterSetName = GetByWorkspaceObjectAndServicePrincipalNameParameterSet,
+            Mandatory = true, HelpMessage = HelpMessages.ServicePrincipalName)]
+        [ValidateNotNullOrEmpty]
+        public string ServicePrincipalName { get; set; }
+
+        [Parameter(ValueFromPipelineByPropertyName = false, ParameterSetName = GetByWorkspaceNameAndIdParameterSet,
+            Mandatory = false, HelpMessage = HelpMessages.PrincipalId)]
+        [Parameter(ValueFromPipelineByPropertyName = false, ParameterSetName = GetByWorkspaceObjectAndIdParameterSet,
+            Mandatory = false, HelpMessage = HelpMessages.PrincipalId)]
+        [Parameter(ValueFromPipelineByPropertyName = false, ParameterSetName = GetByWorkspaceNameAndRoleDefinitionIdAndObjectIdParameterSet,
+            Mandatory = false, HelpMessage = HelpMessages.PrincipalId)]
+        [Parameter(ValueFromPipelineByPropertyName = false, ParameterSetName = GetByWorkspaceObjectAndRoleDefinitionIdAndObjectIdParameterSet,
+            Mandatory = false, HelpMessage = HelpMessages.PrincipalId)]
+        [Alias("Id", "PrincipalId")]
+        [ValidateNotNullOrEmpty]
+        public string ObjectId { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (this.IsParameterBound(c => c.WorkspaceObject))
+            {
+                this.WorkspaceName = this.WorkspaceObject.Name;
+            }
+
+            if (this.IsParameterBound(c => c.RoleDefinitionName))
+            {
+                this.RoleDefinitionId = SynapseAnalyticsClient.GetRoleDefinitionIdFromRoleDefinitionName(this.RoleDefinitionName);
+            }
+
+            if (this.IsParameterBound(c => c.SignInName))
+            {
+                this.ObjectId = SynapseAnalyticsClient.GetObjectIdFromSignInName(this.SignInName);
+            }
+
+            if (this.IsParameterBound(c => c.ServicePrincipalName))
+            {
+                this.ObjectId = SynapseAnalyticsClient.GetObjectIdFromServicePrincipalName(this.ServicePrincipalName);
+            }
+
+            if (this.IsParameterBound(c => c.RoleAssignmentId))
+            {
+                WriteObject(new PSRoleAssignmentDetails(SynapseAnalyticsClient.GetRoleAssignmentById(this.RoleAssignmentId)));
```

Comment:
>) [](start = 124, length = 1)

Do you think whether we should add use `Write-Object(x, true)` just like line #150?

Reply:
It is used for collection scenarios.